tox.ini

```diff
@@ -1,30 +1,60 @@
+# Generated from:
+# https://github.com/zopefoundation/meta/tree/master/config/pure-python
 [tox]
-envlist = flake8,
-          py27,
-          py35,
-          py36,
-          py37,
-          py38,
-          pypy,
-          pypy3,
-          coverage
+envlist =
+    lint,
+    py27,
+    pypy,
+    py35,
+    py36,
+    py37,
+    py38,
+    pypy3,
+    coverage
 
 [testenv]
 usedevelop = true
+deps =
+    zope.testrunner
 commands =
     zope-testrunner --test-path=src []
 extras = test
 
-[testenv:coverage]
-basepython = python3.7
+[testenv:lint]
+basepython = python3
+skip_install = true
+deps =
+    flake8
+    check-manifest
+    check-python-versions
 commands =
-    coverage run -m zope.testrunner --test-path=src []
-    coverage report --fail-under=86
+    flake8 src setup.py
+    check-manifest
+    check-python-versions .
+
+[testenv:coverage]
+basepython = python3
 deps =
     coverage
+    coverage-python-version
+    zope.testrunner
+commands =
+    coverage run -m zope.testrunner --test-path=src []
+    coverage html
+    coverage report -m --fail-under=0
```

Comment:
This had been `--fail-under=86`, please change `.meta.cfg` accordingly so this number is used here.

Reply:
Now, this is surprising and should be mentioned clearly in the documentation.

When I run coverage on the master (without my changes), it fails with 39% - but there not only this package gets tested, but all in site-packages.

I'll have a look how to fix this.

Reply:
When I ran coverage on the current master, as mentioned above, it covers the wrong stuff and even then, this package only has very low test coverage.

```
.tox/coverage/lib/python3.7/site-packages/zope/testrunner/find.py                             231     74    68%
.tox/coverage/lib/python3.7/site-packages/zope/testrunner/formatter.py                        610    405    34%
.tox/coverage/lib/python3.7/site-packages/zope/testrunner/garbagecollection.py                 37     15    59%
.tox/coverage/lib/python3.7/site-packages/zope/testrunner/interfaces.py                        27      0   100%
.tox/coverage/lib/python3.7/site-packages/zope/testrunner/layer.py                             10      3    70%
.tox/coverage/lib/python3.7/site-packages/zope/testrunner/listing.py                           13      5    62%
.tox/coverage/lib/python3.7/site-packages/zope/testrunner/logsupport.py                        20      4    80%
.tox/coverage/lib/python3.7/site-packages/zope/testrunner/options.py                          187     70    63%
.tox/coverage/lib/python3.7/site-packages/zope/testrunner/process.py                           22     12    45%
.tox/coverage/lib/python3.7/site-packages/zope/testrunner/profiling.py                         49     28    43%
.tox/coverage/lib/python3.7/site-packages/zope/testrunner/refcount.py                          61     51    16%
.tox/coverage/lib/python3.7/site-packages/zope/testrunner/runner.py                           668    324    51%
.tox/coverage/lib/python3.7/site-packages/zope/testrunner/selftest.py                           8      0   100%
.tox/coverage/lib/python3.7/site-packages/zope/testrunner/shuffle.py                           23      9    61%
.tox/coverage/lib/python3.7/site-packages/zope/testrunner/statistics.py                        21      2    90%
.tox/coverage/lib/python3.7/site-packages/zope/testrunner/tb_format.py                         52     31    40%
src/zc/zodbrecipes/__init__.py                                                                152    135    11%
src/zc/zodbrecipes/tests.py                                                                    40      3    92%
---------------------------------------------------------------------------------------------------------------
TOTAL                                                                                       22120  13488    39%
ERROR: InvocationError for command /home/jugmac00/Repos/tests/zc.zodbrecipes/.tox/coverage/bin/coverage report --fail-under=86 (exited with code 2)
_________________________________________________________________ summary _________________________________________________________________
ERROR:   coverage: commands failed
```

Reply:
So, no - I won't set coverage to 86.

Reply:
When you have a look at the build history on travis, there was no lint nor coverage runner.

https://travis-ci.org/github/zopefoundation/zc.zodbrecipes/builds

Reply:
Buildout recipes need special magic to measure coverage correctly.  See https://github.com/zopefoundation/z3c.recipe.tag/pull/7

This is documented in [buildout's testing.txt](https://github.com/buildout/buildout/blob/0185dc6a87f417160378547c776fb6da0c016e88/src/zc/buildout/testing.txt#L123-L152).

Reply:
Thanks a lot!

Now, let's see how the config tool likes these changes....

Reply:
Ofc - it does not like the changes... and overwrites them...